pkg/queue/cluster_queue_impl.go

```diff
@@ -32,10 +31,16 @@ type ClusterQueueImpl struct {
 	// across the queues in this ClusterQueue.
 	QueueingStrategy kueue.QueueingStrategy
 
-	heap   heapImpl
+	heap   *heap.Heap
```

Comment:
does it need to be a pointer?

Reply:
no, just personal prefer.

Reply:
In golang, you should prefer values instead of pointers wherever possible, to reduce work for the GC.

Reply:
Changed to struct.